src/cata_tiles.cpp

```diff
@@ -533,34 +555,17 @@ void cata_tiles::draw(int destx, int desty, int centerx, int centery, int width,
     screentile_width = (width + tile_width - 1) / tile_width;
     screentile_height = (height + tile_height - 1) / tile_height;
 
-    for (int my = 0; my < sy; ++my) {
-        for (int mx = 0; mx < sx; ++mx) {
-            x = mx + o_x;
-            y = my + o_y;
-            l = light_at(x, y);
-            const auto critter = g->critter_at( x, y );
-            if (l != CLEAR) {
-                // Draw lighting
-                draw_lighting(x, y, l);
-                if( critter != nullptr && g->u.sees_with_infrared( *critter ) ) {
-                    draw_from_id_string( "infrared_creature", C_NONE, empty_string, x, y, 0, 0 );
-                }
-                continue;
-            }
-            // light is no longer being considered, for now.
-            // Draw Terrain if possible. If not possible then we need to continue on to the next part of loop
-            if (!draw_terrain(x, y)) {
-                continue;
-            }
-            draw_furniture(x, y);
-            draw_trap(x, y);
-            draw_field_or_item(x, y);
-            draw_vpart(x, y);
-            if( critter != nullptr ) {
-                draw_entity( *critter, x, y );
-            }
-        }
-    }
+    auto map_draw_function = [&](int x, int y, lit_level ll) -> void {
+        draw_specific_tile(x, y, ll);
+    };
+
+    g->m.draw_loop(
```

Comment:
The array isn't a compelling problem since there will only be one of them.
'Loop over x and y coordinates' is not a significant piece of code that
needs to be deduplicated.
More importantly, this isn't something that should be married to the draw
code and it shouldn't be re-checked every time we redraw the screen, only
once per turn at the most.
'Additional renderers easier to implement' is also not compelling, for one
thing it doesn't make them any easier since the code in question is trivial
(factoring out the logic for what is visible is however helpful), and more
importantly, the code could change any number of times before that happens
(or it could never happen), so it's not a good rationale.  (See generally
YAGNI)